logstash-core/lib/logstash/modules/kibana_config.rb

```diff
@@ -0,0 +1,103 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging"
+
+require_relative "file_reader"
+require_relative "kibana_resource"
+require_relative "kibana_base_resource"
+
+module LogStash module Modules class KibanaConfig
+  include LogStash::Util::Loggable
+
+  ALLOWED_DIRECTORIES = ["search", "visualization"]
+
+  attr_reader :index_name
+
+  def initialize(modul, settings)
```

Comment:
I left comments for each of the 3 initialize methods:

```
# We name it `modul` here because `module` has meaning in Ruby.
```

I also caught a default credentials error.  It should be `elastic:changeme` rather than `elasticsearch:changeme`.